kod/object/active/holder/room.kod

```diff
@@ -3879,17 +3846,14 @@ messages:
 
    RoomReqCommunication(who=$,type=$,string=$)
    {
-      local oSpell;
-
-      oSpell = Send(SYS,@FindSpellByNum,#num=SID_SILENCE);
+      local lRadiusState;
 
-      if Send(self,@IsEnchanted,#what=oSpell) AND (NOT IsClass(who,&DM))
+      lRadiusState = Send(who,@GetMostPowerfulRadiusEnchantmentState,#byClass=&Silence);
+      if lRadiusState <> $
       {
-         Send(who,@MsgSendUser,#message_rsc=room_silenced_rsc);
-
-         return FALSE;
+         Send(Nth(lRadiusState,1),@ReqCommunication,#who=who,#state=lRadiusState);
```

Comment:
Shouldn't this be returning here based on what ReqCommunication returns to RoomReqCommunication?